client/allocrunner/consulsock_hook.go

```diff
@@ -0,0 +1,319 @@
+package allocrunner
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/nomad/structs/config"
+)
+
+// consulSockHook creates Unix sockets to allow communication from inside a
+// netns to Consul.
+//
+// Noop for allocations without a group Connect stanza.
+type consulSockHook struct {
+	alloc *structs.Allocation
+
+	proxy *sockProxy
+
+	// mu synchronizes group & cancel as they may be mutated and accessed
+	// concurrently via Prerun, Update, Postrun.
+	mu sync.Mutex
+
+	logger hclog.Logger
+}
+
+func newConsulSockHook(logger hclog.Logger, alloc *structs.Allocation, allocDir *allocdir.AllocDir, config *config.ConsulConfig) *consulSockHook {
+	h := &consulSockHook{
+		alloc: alloc,
+		proxy: newSockProxy(logger, allocDir, config),
+	}
+	h.logger = logger.Named(h.Name())
+	return h
+}
+
+func (consulSockHook) Name() string {
```

Comment:
You'll see it in Circle, but this wants to be `func (*consulSockHook) Name() string {`

Reply:
Thanks! Old (bad) habits die hard.